server/aws-lsp-codewhisperer/src/language-server/agenticChat/tools/executeBash.ts

```diff
@@ -0,0 +1,349 @@
+// Port from VSC https://github.com/aws/aws-toolkit-vscode/blob/093d5bcbce777c88cf18c76b52738610263d1fc0/packages/core/src/codewhispererChat/tools/executeBash.ts#L134
+
+import { InvokeOutput } from './toolShared'
+import { split } from 'shlex'
+import { Logging } from '@aws/language-server-runtimes/server-interface'
+import { processUtils } from '@aws/lsp-core'
+import { getUserHomeDir, sanitize } from '@aws/lsp-core/out/util/path'
+
+export enum CommandCategory {
+    ReadOnly,
+    HighRisk,
+    Destructive,
+}
+
+export const dangerousPatterns = new Set(['<(', '$(', '`'])
+export const splitOperators = new Set(['|', '&&', '||', '>'])
+export const splitOperatorsArray = Array.from(splitOperators)
+export const commandCategories = new Map<string, CommandCategory>([
+    // ReadOnly commands
+    ['ls', CommandCategory.ReadOnly],
```

Comment:
Just for reference: this is, at best, an extremely crude heuristic. It can't give any sort of security, it can only help users avoid basic mistakes. 

If we want to go further with the "shell parsing", we'll need a robust library like https://www.shellcheck.net/ . And even then, parsing doesn't actually give guarantees about the binaries on the system or in the user's $PATH.

Reply:
Yeah, I'm hoping this is temporary because it seems like a huge security vulnerability. I will reach out to the original author to confirm.